server/src/test/server-zcap.test.ts

```diff
@@ -0,0 +1,125 @@
+    const createPutSpaceByUuidRequest = (spaceRepresentation: unknown) => new Request(new URL(`/space/${spaceUuid}`, 'http://example.example'), {
+      method: 'PUT',
+      body: JSON.stringify(spaceRepresentation),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+    const spaceToCreate = {
+      controller: keyForAlice.controller
+    }
+    const request = createPutSpaceByUuidRequest(spaceToCreate)
+    const response = await server.fetch(request)
+    assert.equal(response.status, 204, 'response status to PUT /spaces/ MUST be 204')
+
+    await t.test(`GET with no authorization`, async t => {
+      const requestUrl = new URL(`/space/${spaceUuid}`, 'http://example.example')
+      const requestMethod = 'GET'
+      const responseToGetSpace = await server.fetch(new Request(requestUrl, {
+        method: requestMethod,
+        headers: {
+          'Accept': 'application/json',
+        },
+      }))
+      assert.equal(
+        responseToGetSpace.status, 401,
+        'response status to GET /space/:uuid MUST be 401')
+    })
+
+    await t.test(`GET with signature from space controller`, async t => {
+      const requestUrl = new URL(`/space/${spaceUuid}`, 'http://example.example')
+      const requestMethod = 'GET'
+      const responseToGetSpace = await server.fetch(new Request(requestUrl, {
+        method: requestMethod,
+        headers: {
+          'Accept': 'application/json',
+          authorization: await createHttpSignatureAuthorization({
+            signer: keyForAlice,
+            url: requestUrl,
+            method: requestMethod,
+            headers: {},
+            includeHeaders: [
+              '(created)',
+              '(expires)',
+              '(key-id)',
+              '(request-target)',
+            ],
+            created: new Date,
+            expires: new Date(Date.now() + 30 * 1000),
+          }),
+        },
+      }))
+      assert.equal(
+        responseToGetSpace.status, 200,
+        'response status to GET /space/:uuid MUST be 200')
+      const spaceFromGet = await responseToGetSpace.json()
+      assert.equal(spaceFromGet.controller, spaceToCreate.controller)
+    })
+
+    await t.test(`GET with signature over capability-invocation authorized by space controller`, async t => {
+      // introduce a new key that will be delegated to viz zcap
+      const keyForBob = await Ed25519Signer.generate()
+      const requestUrl = new URL(`/space/${spaceUuid}`, 'http://example.example')
+      const requestMethod = 'GET'
+      const capabilityForBobToGetResource = await delegate({
+        signer: keyForAlice,
+        capability: {
+          id: `urn:uuid:${crypto.randomUUID()}`,
+          controller: keyForBob.controller,
+          invocationTarget: urlWithProtocol(requestUrl, 'https:').toString(),
+          allowedAction: [requestMethod],
+          parentCapability: `urn:zcap:root:${encodeURIComponent(urlWithProtocol(requestUrl, 'https:').toString())}`,
```

Comment:
I don't want to have to do `urlWithProtocol(requestUrl, 'https:').toString()`
but it is required for as long as the zcap verifier enforces a requirement of using 'https' and forbids invocationTarget like 'http://localhost'...